### DISCUSSION TITLE
API resources return type magic docblock helper

### DISCUSSION THREAD

**Discussion by phpsa** [dedoc/scramble#661] [Ideas]
If my code in the JsonResource looks like:

```
public function toArray()
{
  return parent::toArray();
}
```
it show `data - string` as the return body,
i have tried docblocks etc but to no avail.

Can we add a doc parser to the resource / toArray method that gets read as a fallback option?

**Comment by romalytvynenko** [comment]
Hey @phpsa 

If you're on the latest version of Scramble, this should work. 

Just make sure you annotate the JSON resource class with the model name: https://scramble.dedoc.co/usage/response#model-resolution

Let me know if this worked for you!

**Reply by phpsa** [reply]
Thanks for the response:
our api does some magic with the response:
so code that looks like:

```
 public function toArray($request)
    {

        return $this->caseFormat($request, [
            'id'                     => $this->id,
            'name'                   => $this->name,
            'file_url'               => $this->file_url,
            'thumbnail_url'          => $this->thumbnail_url,
            'ice_product_id'         => $this->ice_product_id,
            'ice_schema_id'          => $this->ice_schema_id,
            'is_public'              => $this->is_public,
            'display_dealer_default' => $this->display_dealer_default,
            'type'                   => $this->type,
            'created_at'             => $this->created_at,
            'updated_at'             => $this->updated_at,
        ]);

      //  return  $this->caseFormat($request, $data);
    } 
```
looks like:
![image](https://github.com/user-attachments/assets/ed00eafd-e335-4129-96d7-508bbf40248f)

and some of the fields are dynamic aswell.
was hoping could do something along the lines of a fallback so that could:

```
/**
* @return array{id:int, name:string, file_url?:?string}
*/
 public function toArray($request)
    {
    parent::toArray();
    }
```
for
![image](https://github.com/user-attachments/assets/c797b6dd-1831-4fc5-9d6a-5a3976ae306f)


preferably following phpstan rules for the return types (else happy to do a @response version if needed.
